src/artm/core/processor.cc

```diff
@@ -57,14 +57,16 @@ class RegularizeThetaAgentCollection : public RegularizeThetaAgent {
 
   bool empty() const { return agents_.empty(); }
 
-  virtual void Apply(int item_index, int inner_iter, int topics_size, const float* n_td, float* r_td) const {
+  virtual void Apply(const std::string& item_title, int inner_iter, int topics_size,
+                     const float* n_td, float* r_td) const {
     for (auto& agent : agents_)
-      agent->Apply(item_index, inner_iter, topics_size, n_td, r_td);
+      agent->Apply(item_title, inner_iter, topics_size, n_td, r_td);
   }
 
-  virtual void Apply(int inner_iter, const LocalThetaMatrix<float>& n_td, LocalThetaMatrix<float>* r_td) const {
+  virtual void Apply(int inner_iter, const LocalThetaMatrix<float>& n_td,
+                     LocalThetaMatrix<float>* r_td, const Batch& batch) const {
     for (auto& agent : agents_)
-      agent->Apply(inner_iter, n_td, r_td);
+      agent->Apply(inner_iter, n_td, r_td, batch);
```

Comment:
ThetaAgents exists for performance reasons -- processor.cc calls Regularizer.CreateAgent for each new batch. At this moment regularizer can do any "slow" processing (once per batch), and then agents can be applied quickly in the innermost loop (e.g. when processing given document and inner iteration). 
I reverted changes here:
https://github.com/ofrei/bigartm/commit/b2a7ec72ead61a6168bb840275535d7c7d9ed7e6
And while doing this I found a big issue. You've changed a signature of RegularizerInterface::Apply, but didn't change some implementation --- `hierarchy_sparsing_theta`, `smooth_ptdw`, `topic_segmentation_ptdw`. That's not good :) Could this be a reason for python tests to fail?